### ISSUE TITLE
Outstanding symlink issues

### ISSUE THREAD

**Issue by ddavness** [roblox-ts/roblox-ts#928]
**Note:** Since these issues only affect a niche of people (me), I'm willing to provide the fixes myself...

## Version

**roblox-ts:** master branch (0.3.1)

**@rbxts/types:** N/A

## Description

At the moment, the way roblox-ts performs some of it's I/O operations might be destructive in some non-conventional directories (in particular, symbolic links):

### 1. Initial compile would overwrite symlinks (Fixed by #921)
When doing the initial compile (basically first compilation when `-w` is provided, but this also happens regardless), all the files from `rootDir` are copied (overwritten) to `outDir`. However, this also happens to directories. On folders it isn't a problem, but on symlinks it is. See the mentioned pull request for more info.

### 2. Directory renaming can cause undesired behavior if the root/target directory is a symlink (Fixed by #938)
Better explained if I use an example:

![image](https://user-images.githubusercontent.com/45321798/74068011-701bba00-49f2-11ea-8a4e-f0e3d2872967.png)

`src` is `rootDir`
`lua` is `outDir`

`src/global` and `lua/global` are symlinks.
Should we rename `src/global` to `src/global_2` while watching (`rbxtsc -w`), we get the following output:

```
remove C:\Users\davness\Documents\Git\roblox-ts-skeleton\Places\place\lua\global\test.lua
remove C:\Users\davness\Documents\Git\roblox-ts-skeleton\Places\place\lua\global
(node:15708) UnhandledPromiseRejectionWarning: Error: EPERM: operation not permitted, symlink 'C:\Users\davness\Documents\Git\roblox-ts-skeleton\Places\global\src' -> 'C:\Users\davness\Documents\Git\roblox-ts-skeleton\Places\place\lua\global_2'
(node:15708) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
(node:15708) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

Other side problems we get is that because we end up deleting everything inside the symlink, and therefore we also end up deleting the stuff within the target directory. In this case, it's a folder shared across places, so we don't want this:

In conclusion:

**What happens:** Stuff inside the directory is *deleted* (and so affects the external environment), the watcher ends up erroring.

**What should happen:** At most, only the symlink (not what's inside), should be deleted; Instead of trying to copy the symlink (which will lead to an error, in this case), just create a folder with the same name and proceed from there (since chokidar doesn't support renames),
